src/coreclr/inc/debugmacros.h

```diff
@@ -56,8 +55,12 @@ extern VOID ANALYZER_NORETURN DbgAssertDialog(const char *szFile, int iLine, con
 
 #else // !_DEBUG
 
-#define _ASSERTE(expr) ((void)0)
```

Comment:
I would just not define our ones for debug and leave these alone.

Reply:
I changed this as requested and it broke CI. I'm guessing I should patch the individual broken files? It builds locally for me.

Reply:
It's actually a clash between PAL assert and debugmacros, it's not clear how I would fix it without doing the ifndef.

Reply:
Reverting back to the ifndef for now. Let me know if there's something you'd like to see me do here - I didn't see any easy path towards cleaning up the clash between the PAL and debugmacros other than an ifndef or undef.